From 03062d0d9d9fe229f2ff55d600b70ff64b27edde Mon Sep 17 00:00:00 2001
From: Jeremias Ortega <jeremias@ortega.tech>
Date: Tue, 14 Jul 2020 04:38:25 +0000
Subject: [PATCH] extensionDownloader: Fix check for updates with several
 extensions

When having several extensions installed checking for updates fails.
This is because we are using GET and query params and since we are
sending all the metadata of the extension the server returns 502
when the URL is too long. This error code is ignored safely.

It is only needed to send the version of the extension to check if it
has updates.

Fixes https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2962
https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1363


(cherry picked from commit f0d2509dc34b88eedba43e4537d0d4e837347d54)
---
 js/ui/extensionDownloader.js | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/js/ui/extensionDownloader.js b/js/ui/extensionDownloader.js
index 90fc5a6161..a7b40f9735 100644
--- a/js/ui/extensionDownloader.js
+++ b/js/ui/extensionDownloader.js
@@ -139,7 +139,9 @@ function checkForUpdates() {
             return;
         if (extension.hasUpdate)
             return;
-        metadatas[uuid] = extension.metadata;
+        metadatas[uuid] = {
+            version: extension.metadata.version,
+        };
     });
 
     if (Object.keys(metadatas).length === 0)
-- 
GitLab

